daemon/cluster/cluster.go

```diff
@@ -1075,10 +1075,13 @@ func (c *Cluster) RemoveNetwork(input string) error {
 	return nil
 }
 
-func populateNetworkID(ctx context.Context, c swarmapi.ControlClient, s *types.ServiceSpec) error {
+func (c *Cluster) populateNetworkID(ctx context.Context, client swarmapi.ControlClient, s *types.ServiceSpec) error {
 	for i, n := range s.Networks {
-		apiNetwork, err := getNetwork(ctx, c, n.Target)
+		apiNetwork, err := getNetwork(ctx, client, n.Target)
 		if err != nil {
+			if ln, _ := c.config.Backend.FindNetwork(n.Target); ln != nil && !ln.Info().Dynamic() {
```

Comment:
I know. I was trying to keep the impact to the minimal by not changing the core functions.
For example, today we have a clear separation between swarm's getNetwork and daemon's FindNetwork. There is no single place today one can get all the networks.

The better fix would be to come up with a elegant function with appropriate filters and context that will differentiate the source of truth & determine eligibility. But I didnt want to invest the effort atm.